pkg/telemetry/phonehome/interceptor.go

```diff
@@ -45,11 +46,28 @@ func getUserAgent[getter func(string) []string](headers getter) string {
 	// https://github.com/grpc/grpc-go/blob/0238b6e1cec37b55820b461d3d30652c54efe2c4/clientconn.go#L211-L215
 	userAgentValues := headers(grpcGatewayUserAgentHeader)
 	// If endpoint is accessed not via grpc-gateway, extract from User-Agent header.
-	// If endpoint is accessed via grpc-gateway, append grpc-go value to the resultinguser agent.
 	userAgentValues = append(userAgentValues, headers("User-Agent")...)
 	return strings.Join(userAgentValues, " ")
 }
 
+// Metadata stores and accesses the headers in lowercase. This is incompatible
+// with http.Header, which uses canonized keys.
+type withSimpleGet metadata.MD
+
+// Get implements the Getter interface for gRPC metadata to work similarly to
+// the http.Header.Get, that returns the first value from the array.
+// The grpcgateway
+func (md withSimpleGet) Get(key string) string {
+	// Add grpcgateway- prefix to the key if the request came via grpc-gateway.
+	if len((metadata.MD)(md).Get(runtime.MetadataPrefix+"Accept")) > 0 {
```

Comment:
```suggestion
  m := (metadata.MD)(md)
	if len(m.Get(runtime.MetadataPrefix+"Accept")) > 0 {
```

Reply:
Or maybe use https://github.com/grpc-ecosystem/go-grpc-middleware/blob/v1.4.0/util/metautils/nicemd.go#L85

Reply:
or better https://github.com/grpc-ecosystem/go-grpc-middleware/blob/3834477040993ff385d88890844cf851e84b01d2/metadata/metadata.go#L85

Reply:
Updated in favor of `requestinfo` refactoring.